src/_data/glossary.yml

```diff
@@ -1,3 +1,20 @@
+- term: "Analysis context"
+  short_description: |-
+    A representation of a set of code that should be analyzed using the same
+    sources of metadata.
+  long_description: |-
+    An _analysis context_ is a representation of a set of code that should be
```

Comment:
I'm not sure I like "set of code." (Though I'm game to use consistent terms wherever we can, if it's a consistent term.)

In my mind this conjures an image of a pile of subroutines, like a 30-line function here and a 100-line class there.

One property that I think would be helpful for users is that the "set of code" is a set of files that are more-or-less co-located in the filesystem. So you have properties like:

1. All of the files in one directory are always in the same analysis context.
2. All of the files in one directory are in the same analysis context as the files in any subdirectories, unless there exists a new "metadata" file (`.dart_tool/package_config.json`). (Hard to write that one concisely and without explicitly introducing the package config file that we don't want users to worry about.)

In practice, as you look at a directory tree of source files, a new analysis context is introduced each time you see a `pubspec.yaml` file, as those generally pair with generated `.dart_tool/package_config.yaml` files. But if you haven't run `dart pub get` yet, that's not really true. And if any pubspec file is using the new Workspaces feature, it's not really true again.

After all that, I think the only suggestion I have is to use something like "set of Dart source files" rather than "set of code." Again, consistency with other text being important.

Reply:
> ... without explicitly introducing the package config file that we don't want users to worry about.

I disagree. I think it's important for users to understand the role of the package config file and its relationship to a `pubspec.yaml` file (or potentially multiple `pubspec.yaml` files in the case of workspaces. It's essential to understanding how absolute URIs in directives are resolved.